agent-core/src/main/java/com/newrelic/agent/android/instrumentation/HttpURLConnectionExtension.java

```diff
@@ -576,14 +577,17 @@ void addTransactionAndErrorData(TransactionState transactionState) {
                 params.put(Constants.Transactions.CONTENT_TYPE, contentType);
             }
 
-			/* Also add Content-Length. TransactionState bytesReceived is set directly
+            /* Also add Content-Length. TransactionState bytesReceived is set directly
              * from the Content-Length header in this error case.
-			*/
+             */
             params.put(Constants.Transactions.CONTENT_LENGTH, transactionState.getBytesReceived() + "");
 
             transactionData.setResponseBody(responseBody);
-            transactionData.setParams(params);
-
+            if (transactionData.getParams() != null) {
```

Comment:
Why not just alloc the `params` field in the ctor, rather than duplicate this nullability check each place it's used?

Reply:
got it.